File: LeetCode/2357. Make Array Zero by Subtracting Equal Amounts/Solution.java
/**
 * Runtime: 1 ms
 * Memory Usage: 39 MB 
 */
class Solution {
    public int minimumOperations(int[] nums) {
        HashSet<Integer> rec = new HashSet<>();
        for (int num: nums) rec.add(num);
        return rec.contains(0) ? rec.size() - 1 : rec.size();
    }
}

File: LeetCode/2357. Make Array Zero by Subtracting Equal Amounts/solution.go
/**
 * Runtime: 0 ms
 * Memory Usage: 2 MB 
 */
func minimumOperations(nums []int) int {
    rec := make(map[int]struct{})
    for _, num := range nums {
        if num > 0 {
            rec[num] = struct{}{}
        }
    }
    return len(rec)
}
